aspnetcore/mvc/controllers/routing/samples/6.x/main/Controllers/TestController.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+public class TestController : Controller
+{
+    #region snippet
+    public IActionResult Index()
+    {
+        var url = Url.Action("Buy", "Products", new { id = 17, color = "red" });
+        return Content(url);
```

Comment:
@IEvangelist what's the cleanest way to address warning CS8604: Possible null reference argument for parameter 'content' in 'ContentResult ControllerBase.Content(string content)'.

Reply:
In this case, the `ControllberBase.Content(string content)` is specifying that it's parameter is non-nullable. But it appears that the `Url.Action` can return `null`? If that is true, but you're certain it will not be `null` here - you should tell the compiler that by using the null-forgiving operator `!` (sometimes called the damnit operator).

```csharp
return Content(url!);
```

Reply:
```suggestion
        var url = Url.Action("Buy", "Products", new { id = 17, color = "red" });
        return Content(url!);
```